src/Xamarin.Android.Build.Tasks/Tasks/CompileToDalvik.cs

```diff
@@ -120,12 +127,12 @@ protected override string GenerateCommandLineCommands ()
 					var zip = Path.GetFullPath (Path.Combine (ClassesOutputDirectory, "..", "classes.zip"));
 					if (File.Exists (zip)) {
 						Log.LogDebugMessage ($"    {zip}");
-						sw.WriteLine (Path.GetFullPath (zip));
+						sw.WriteLine (Path.GetFullPath (zip).Normalize (NormalizationForm.FormC));
 					}
 					foreach (var jar in JavaLibrariesToCompile) {
 						var fullPath = Path.GetFullPath (jar.ItemSpec);
 						Log.LogDebugMessage ($"    {fullPath}");
-						sw.WriteLine (fullPath);
+						sw.WriteLine (fullPath.Normalize (NormalizationForm.FormC));
```

Comment:
Assuming my prior musings are remotely accurate, this change should *not* be necessary.  `JavaLibariesToCompile` comes from filesystem reads (`<DetermineJavaLibrariesToCompile/>` and co), and thus should be assumed to have the correct filesystem encoding/(non-)normalization.